clang/test/SemaCXX/warn-unsafe-buffer-usage-array.cpp

```diff
@@ -38,3 +38,17 @@ void constant_idx_unsafe(unsigned idx) {
                         // expected-note@-1{{change type of 'buffer' to 'std::array' to label it for hardening}}
   buffer[10] = 0;       // expected-note{{used in buffer access here}}
 }
+
+void constant_id_string(unsigned idx) {
+  char safe_char = "abc"[1]; // no-warning
+  safe_char = ""[0];
+  safe_char = "\0"[0];
+ 
+  char abcd[5] = "abc";
+  abcd[2]; // no-warning
+
+  char unsafe_char = "abc"[3]; //expected-warning{{unsafe buffer access}}
```

Comment:
The null at the end is still within bounds. This case should not be a warning.

Reply:
Okay. Makes sense.